tests/vxlan/test_vxlan.py

```diff
@@ -0,0 +1,645 @@
+
+    return return_dict
+
+def create_vnets(duthost, tunnel_name, vnet_count=1, scope=None, vni_base=10000, vnet_name_prefix="Vnet"):
+    return_dict = {}
+    scope_entry = ""
+    if scope:
+        scope_entry = '''"scope": "{}",'''.format(scope)
+    config_list = []
+    for i in range(vnet_count):
+        name = vnet_name_prefix + "-" + str(i)
+        vni = vni_base+i
+        return_dict[name] = vni
+        config_list.append('''"{}": {{
+               "vxlan_tunnel": "{}",
+               {}"vni": "{}",
+               "peer_list": ""
+        }}'''.format(name, tunnel_name, scope_entry, vni))
+
+        full_config = '{\n"VNET": {' + ",\n".join(config_list) + '\n}\n}'
+
+    apply_config_in_dut(duthost, full_config, "vnets_"+tunnel_name)
+    return return_dict
+
+def setup_vnet_intf(duthost, selected_interfaces, vnet_list, minigraph_data):
+    if len(selected_interfaces) != len(vnet_list):
+        raise RuntimeError("Different number of interfaces and vnets, not supported yet")
+
+    ret_list = {}
+    intf_config_list = []
+    po_config_list = []
+    for count in range(len(selected_interfaces)):
+        intf = selected_interfaces[count]
+        config = ('''
+                "{}" : {{
+                    "vnet_name": "{}"
+                }}
+        '''.format(intf, vnet_list[count]))
+
+        if "Ethernet" in intf:
+            intf_config_list.append(config)
+            ret_list[intf] = vnet_list[count]
+        elif "PortChannel" in intf:
+            po_config_list.append(config)
+            for member in get_ethernet_ports([intf], minigraph_data):
+                ret_list[member] = vnet_list[count]
+
+    full_config_list = []
+    if intf_config_list:
+        full_config_list.append(
+            '''"INTERFACE": {\n''' + ",\n".join(intf_config_list) + '''}''')
+    if po_config_list:
+        full_config_list.append(
+            '''"PORTCHANNEL_INTERFACE": {\n''' + ",\n".join(po_config_list) + '''}''')
+
+    full_config = '''{\n''' + ",\n".join(full_config_list) + '''}'''
+    apply_config_in_dut(duthost, full_config, "vnet_intf")
+    return ret_list
+
+def configure_vxlan_switch(duthost, vxlan_port=4789, dutmac=None):
+    if dutmac == None:
+        #dutmac = duthost.facts['router_mac']
+        dutmac = "aa:bb:cc:dd:ee:ff"
+
+    switch_config = '''
+[
+        {{
+                "SWITCH_TABLE:switch": {{
+                        "vxlan_port": "{}",
+                        "vxlan_router_mac": "{}"
+                }},
+                "OP": "SET"
+        }}
+]
+'''.format(vxlan_port, dutmac)
+    apply_config_in_swss(duthost, switch_config, "vnet_switch")
+
+def apply_config_in_swss(duthost, config, name="swss_"):
+    if Constants['DEBUG']:
+        filename = name + ".json"
+    else:
+        filename = name + "-" + str(time.time()) + ".json"
+
+    duthost.copy(content=config, dest="/tmp/{}".format(filename))
+    duthost.shell("docker cp /tmp/{} swss:/".format(filename))
```

Comment:
Thanks for this command. I got this logic from the older scripts. I will implement the way you have suggested. Also the shell() function itself raises exception in case of failure. So there is no need to check the result.